lib/ansible/galaxy/collection.py

```diff
@@ -584,6 +591,82 @@ def build_collection(collection_path, output_path, force):
+        )
+
+        b_collection_path = to_bytes(os.path.join(search_path, namespace, name), errors='surrogate_or_strict')
+
+        if not os.path.isdir(b_collection_path):
+            continue
+
+        if not os.path.isfile(os.path.join(b_collection_path, b'MANIFEST.json')) and not fallback_metadata:
+            non_artifact_found = True
+            continue
+
+        local_collection = CollectionRequirement.from_path(b_collection_path, False, fallback_metadata=fallback_metadata, parent=parent)
+
+        if not _collection_matches_requirement(local_collection, requirement, parent):
+            continue
+
+        found = True
+        display.vvvv("Found collection '%s.%s:%s'" % (namespace, name, requirement))
+        yield local_collection
+        if first_found:
+            break
+
+    if require_match and not found:
+        if not fallback_metadata and non_artifact_found:
+            raise AnsibleError(
+                message="Collection %s.%s does not appear to have a MANIFEST.json. " % (namespace, name) +
+                "A MANIFEST.json is expected if the collection has been built and installed via ansible-galaxy.")
+        else:
+            raise AnsibleError(message='Collection %s.%s is not installed in any of the collection paths.' % (namespace, name))
+    elif not found:
+        display.display('%s.%s:%s is not installed' % (namespace, name, requirement))
+
+
+def remove_collections(collections, search_paths, ignore_errors, no_deps, parent=None):
+    with _display_progress():
+        for collection in collections:
+            b_collection = to_bytes(collection[0], errors='surrogate_or_strict')
+            if os.path.isfile(b_collection) or urlparse(collection[0]).scheme.lower() in ['http', 'https'] or len(collection[0].split('.')) != 2:
+                raise AnsibleError(message="'%s' is not a valid collection name. The format namespace.name is expected." % collection[0])
+
+            collection_name = collection[0]
+            namespace, name = collection_name.split('.')
+            collection_version = collection[1]
+
+            # do we want to remove all or first found? should it be a toggle?
+            try:
+                found_collections = find_matching_collections(namespace, name, collection_version, search_paths, fallback_metadata=True, parent=parent)
+                for local_collection in found_collections:
+                    local_collection.remove()
+
+                    # if an error is encountered above but --ignore_errors is set should dependencies be removed anyway instead?
+                    if not no_deps:
+                        deps = [[identifier, requirement] for identifier, requirement in local_collection.dependencies.items()]
+                        remove_collections(deps, search_paths, ignore_errors, no_deps, parent=local_collection)
```

Comment:
We have to be careful that we don't remove deps of another collection, i.e.

* community.general depends on ansible.posix
* other.collection also depends on ansible.posix

If we were to remove `community.general` then `ansible.posix` should not be removed because another collection relies on it.

Reply:
Good point!

Reply:
I've fixed this, let me know what you think.